### ISSUE TITLE
kernel error i8042: No Controller Found on MacBookPro4,1 (A1260) while booting Elementary from USB stick

### ISSUE THREAD

**Issue by Binker** [aroman/freya-on-a-mac#30]
Hi,

Like the subject says, I'm getting the i8042: No Controller Found error while booting into Elementary from a USB stick. I've installed rEFInd with ./install.sh --notesp --alldrivers and I get into the boot menu of Elementary (try/install/oem,etc...). I'm currently experimenting with different boot options (acpi_osi= / acpi=off / nomodeset / etc), going through them one by one hoping that one or the other resolves the issue. The error refers to (as far as I've been able to figure out) to support for PS2 keyboards or the display controller - which one still remains unclear to me, simply because the error msg isn't very clear. 

However, I'm currently running the Mac without a battery because it got overheated at one point, and started bulging (this happend to a 2006 Mac I have as well). Could this be the reason for the error msg?

One last thing; you say that you assume that the Mac "must not have a Core 2 duo [...] processor...". Care to explain why?? 
Please note that my spesific model DOES have 64-Bit EFI Architecture.
I've checked both Everymac.com (http://www.everymac.com/systems/apple/macbook_pro/specs/macbook-pro-core-2-duo-2.5-15-early-2008-penryn-specs.html#macspecs1) and manually $ ioreg -l -p IODeviceTree | grep firmware-abi  (returns: "firmware-abi" = <"EFI64">)
I realise that older models use 32-bit EFI, but if there's another reason for not being able to run Elementary on the 2008-models, please let me know.

Thanks!!

/Binker

**Comment by sdaitzman** [comment]
I believe I checked several Core 2 Duo laptops and found that they almost all had 32-bit EFI. If yours has 64-bit, I do believe it should work!

The `--notesp` flag is almost definitely not one you want to use. Can you install rEFInd again without that flag? It installs rEFInd to a less-than-ideal location.

**Comment by Binker** [comment]
Thanks for answering! 
Yes, the 2008-models have 64-bit EFI, earlier than that i believe they're all 32-bit. 

I tried without the --notesp flag the first time around, but couldn't get it to find the usb stick. Then I reinstalled with the options I stated in my initial post, and then I could find "EFI" when holding down Alt at boot. 

However, I've done something - what, exactly, I'm not sure. Yesterday, I saw "Macintosh HD", "EFI" and "Recovery HD" when holding down Alt. Today I've got two "EFI's" and "Recovery", but no "Mac HD". Still, If I choose "EFI", I can choose to boot on the "Mac HD" after entering into the menu of rEFInd. 

I didn't pay much attention to those things, though, since I'm able to boot from the USB stick. My main concern is now that I get "(initramfs) unable to find a medium containing a live file system" quite "a long while" (if that's even an expression) into the booting process. Setting the acpi=off seemed to solve my initial problem with the i8042 error msg. I mean, I still get it, but the boot process doesn't freeze anymore.

**Comment by pefferen** [comment]
There seem to be issues with the usb3 drivers , you could try creating a bootavle SD card instead. That is hoe I  installed linux on my macbook pro.

**Comment by Binker** [comment]
Thanks for the tip! I've tried with quite a lot of boot options now, and it does indeed seems like it looses track of where the install media is mounted so usb3 drivers seems like a good candidate for the source of the problem. I don't have any SD cards lying around (that I know of, need to look around a little), so I was looking into burning a live CD. It's the summer holidays now, so I'll report back when I find the time to check it out.

**Comment by Binker** [comment]
Turned out I had a little time to spare, and burned a CD. Installing as I'm writing this!!  :-)
I did get a brief notification on screen saying "i8042: No Controller Found", but it loaded the splash screen with the Elementary logo right after that. In addition, I couldn't get it to connect to wifi, but I'll update via Ethernet cable to see if that does anything. Important thing is: I "bypassed" the error by using a CD instead of USB, which again leads to the conclusion that there's an issue with the usb3 drivers. Thanks for the help, people!
